prometheus_client/samples.py

```diff
@@ -32,12 +32,17 @@ def __gt__(self, other):
 
 
 # Timestamp and exemplar are optional.
-# Value can be an int or a float.
+# Value must be a float.
```

Comment:
Actually this is a problem, we must support ints too. If someone passes in an int, it needs to stay as an int.

Reply:
Had a go at this, split over two commits, what do you think?